app/models/spree/product_import.rb

```diff
@@ -1,602 +1,620 @@
-          end
-        end
-      end
-    end
-    ### END TAXON HELPERS ###
-
-    # May be implemented via decorator if useful:
-    #
-    #    Spree::ProductImport.class_eval do
-    #
-    #      private
-    #
-    #      def after_product_built(product, params_hash)
-    #        super()
-    #        # do something with the product
-    #      end
-    #    end
-    def after_product_built(product, params_hash)
-      if (params_hash[:locale])
-        add_translations(product, params_hash)
-      end
-    end
-
-    # TODO: Translate slug
-    def add_translations(product, params_hash)
-      localeProduct=params_hash[:locale]
-      if (localeProduct.nil?) then return end
-
-      translations_names=product.translations.attribute_names
-      #product_fields=product.attribute_names
-      #Necesitamos "duplicar" el campo (de momento, slug) que nos sirve para detectar
-      #si el producto existe. Por tanto, si estamos traduciendo este campo, lo tendremos
-      #en dos columnas del csv. En una con el valor original, i en otra donde pondremos
-      #la traducción.
-      # if (params_hash.include?(ProductImport.settings[:variant_comparator_field_i18n]) )
-      #   translations_names.delete(ProductImport.settings[:variant_comparator_field].to_s)
-      #   #translations_names << ProductImport.settings[:variant_comparator_field_i18n].to_s
-      # end
-      translation=product.translations.where(locale: localeProduct).first_or_initialize
-      params_hash.each do |key,value|
-        if translations_names.include?(key.to_s)
-          #Detectamos si el campo és el slug traducido, y en tal caso, lo añadimos
-          #con el nombre "slug"
-          #if (key.to_s==ProductImport.settings[:variant_comparator_field_i18n].to_s)
-          #translation.send("#{ProductImport.settings[:variant_comparator_field].to_s}=", value)
-          #product.attributes={ProductImport.settings[:variant_comparator_field].to_s => value, :locale => localeProduct}
-          #else
-          translation.send("#{key.to_s}=", value)
-          #end
-        end
-      end
-      translation.save
-    end
-    #Special process of prices because of locales and different decimal separator characters.
-    #We want to get a format with dot as decimal separator and without thousand separator
-    def convertToPrice(priceStr)
-      punt=priceStr.index('.')
-      coma=priceStr.index(',')
-      #If the string contains dot and commas, we process it. If not, we replace comma by dot.
-      if (coma!=nil && punt!=nil)
-        #If dot is before comma, the format is x.xxx,xx so we delete the dot.
-        #If not, the format is x,xxx.xx so we delete the comma.
-        if (punt<coma)
-          priceStr=priceStr.gsub('.', '')
-        else
-          priceStr=priceStr.gsub(',', '')
-        end
-      end
-      #We replace comma by dot.
-      return priceStr.gsub(',', '.').to_f
-    end
-  end
-end
+# This model is the master routine for uploading products
```

Comment:
changing the file format with the other changes, made this difficult to review.

I'm going to made the format change in other PR.